File: compiler/engine/testdata/reference/control_flow/engine.ts
// TypeScript equivalent of control_flow/engine.go.
import { Archetype, entityMemory, defineImport } from '@sonolus/sonolus.js-compiler/play'

const archetype = new Archetype('N')
const count = entityMemory(Number)
const flag = entityMemory(Number)
defineImport(count)
defineImport(flag)

export function initialize() {
	count.set(0)
	flag.set(0)
}

export function updateSequential() {
	for (let i = 0; i < 5; i++) {
		if (i > 2) {
			count.set(count.get() + 1)
		}
	}
	if (count.get() > 3) {
		flag.set(1)
	}
}

File: internal/compiler/engine/testdata/reference/control_flow/engine.go
package p

type N struct {
	Count float64 `sonolus:"memory"`
	Flag  float64 `sonolus:"memory"`
}

func (n N) Initialize() {
	n.Count = 0
	n.Flag = 0
}

func (n N) UpdateSequential() {
	for i := 0; i < 5; i++ {
		if i > 2 {
			n.Count = n.Count + 1
		}
	}
	if n.Count > 3 {
		n.Flag = 1
	}
}
